### DISCUSSION TITLE
What is the purpose of maintaining the audit details for each document in the database collections?

### DISCUSSION THREAD

**Discussion by agrahari-himanshu** [olasunkanmi-SE/restaurant#280] [Q&A] [answered]
Hi,

I was wondering what would be the purpose of maintaining the audit info for each document created?
I know it is important for some of the schema to track the audit info, but not for all the documents

```ts
export interface IAudit {
  readonly auditCreatedDateTime: string;
  readonly auditCreatedBy: string;
  auditModifiedBy?: string;
  auditModifiedDateTime?: string;
  auditDeletedBy?: string;
  auditDeletedDateTime?: string;
}
```
Can someone help me to understand it's use case?

**Note:** I have seen we keep mostly only createdAt and updatedAt timestamps by default, not this much details
Thanks

**Comment by olasunkanmi-SE** [comment, marked as answer]
The audit is required for any model within an application because it gives us information on who did what and when. Therefore the need for other properties such as auditModifiedDateTime, auditDeletedBy, and auditDeletedDateTime. Generally, it is a way to keep track of user activities within the application. @HimanshuAgrahari07

**Reply by agrahari-himanshu** [reply]
Thanks for the information, that makes sense to some extent and completely true to track those information.
